cmake/dts.cmake

```diff
@@ -55,6 +55,11 @@ set(dts_files
   ${shield_dts_files}
   )
 
+# Pass preprocessor flags from the environment
+set(env_dts_cpp $ENV{DTS_EXTRA_CPPFLAGS})
```

Comment:
This takes settings from environment.
Why is `-DDTS_EXTRA_CPPFLAGS=<flags>` not sufficient ?

In general, Zephyr doesn't allow for passing compiler flags through environment.
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/generic_toolchain.cmake#L16-L17

> message(WARNING "The environment variable '${var}' was set to $ENV{${var}},
but Zephyr ignores flags from the environment. Use 'cmake -DEXTRA_${var}=$ENV{${var}}' instead.")

Reply:
When I searched for how to do this I found examples using this idiom, so I followed it.

There are cases where I'd want to add or set flags in a `CMakeLists.txt` file, as well as externally.  I'll see if I can come up with a way that doesn't involve the environment.

Reply:
> When I searched for how to do this I found examples using this idiom, so I followed it.

For conf files, I believe, not for preprocessing / compile flags.